File: mobile/SilniaPreview1/app/src/main/java/com/example/silniapreview1/Silnia.kt
package com.example.silniapreview1

class Silnia {
    public fun Iter(n:Int):Int{
        if(n<=1) return 1;
        var result = 1;
        for (i in 1..n){
            result *= i;
        }
        return result
    }
    public fun Rec(n:Int):Int {
        if (n<=1) return 1;
        return Rec(n-1)*n;
    }
}

File: web/cw14-silnia/src/Utility/Silnia.ts
export class Silnia {
  public static Iteration(n: number): number {
    if (n <= 1) return 1;
    let result = 1;
    for (let i = 1; i <= n; i++) {
      result *= i;
    }
    return result;
  }
  public static Recursion(n: number): number {
    if (n <= 1) return 1;
    return this.Recursion(n - 1) * n;
  }
}
